core/src/banking_stage/transaction_scheduler/receive_and_buffer.rs

```diff
@@ -430,6 +430,7 @@ impl TransactionViewReceiveAndBuffer {
                     if result.is_err() {
                         num_dropped_on_status_age_checks += 1;
                         container.remove_by_id(priority_id.id);
+                        continue;
                     }
                     let transaction = &container
                         .get_transaction_ttl(priority_id.id)
```

Comment:
without the `continue` above, the transaction is no longer in the container; the `expect` will always fail if the `result` is an error here.